editors/code/.eslintrc.js

```diff
@@ -1,38 +1,52 @@
+// @ts-check
+
+/** @type { import('eslint').Linter.Config } */
 module.exports = {
-    "env": {
-        "es6": true,
-        "node": true
+    env: {
+        es6: true,
+        node: true,
     },
-    "parser": "@typescript-eslint/parser",
-    "parserOptions": {
-        "project": "tsconfig.json",
-        "sourceType": "module"
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+        project: 'tsconfig.json',
+        sourceType: 'module',
     },
-    "plugins": [
-        "@typescript-eslint"
+    plugins: ['@typescript-eslint'],
+    extends: [
+        'eslint:recommended',
```

Comment:
We intentionally don't use recommended: https://github.com/rust-analyzer/rust-analyzer/issues/3041

Reply:
We use it, but the irrelevant rules are explicitly turned off down bellow.